File: frontend/lib/features/authentication/data/user_model.dart
class UserModel {
  final String id;
  final String username;
  final String emailId;
  final String refreshToken;
  final bool isLoggedIn;

  UserModel({
    required this.id,
    required this.username,
    required this.emailId,
    required this.refreshToken,
    required this.isLoggedIn,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json["id"],
      username: json["username"],
      emailId: json["emailId"],
      refreshToken: '',
      isLoggedIn: json["isLoggedIn"] ?? false
    );
  }

  factory UserModel.fromLoginJson(Map<String, dynamic> json) {
    final user = json["user"];
    return UserModel(
      id: user["id"],
      username: user["username"],
      emailId: user["emailId"],
      refreshToken: json["refreshToken"],
      isLoggedIn: user["isLoggedIn"]
    );
  }

  Map<String, dynamic> toMap() {
    return {'id': id, 'username': username, 'emailId': emailId,  'refreshToken': refreshToken,
    'isLoggedIn': isLoggedIn
    };
  }
}

File: backend/src/model/userModel.js
const mongoose = require('mongoose');
const { authDB } = require('../config/dbConnection');

const userSchema = mongoose.Schema({
    username : {
        type:String,
        required: [true,"Please enter Username"]
    },

    emailId : {
        type:String,
        unique: true,
        required: [true,"Please enter emailID"]
    },

    password: {
        type: String,
        required: [true,"Please enter your password"]
    },

    phoneNumber: {
        type:String,
    },

    role:{
        type: String,
        enum: ["User","Admin"],
        default: "User"
    },

    isLoggedIn:{
        type: Boolean
    },

    refreshToken:{
        type:String
    }
});

const User = authDB.model('User', userSchema); 
module.exports = User;
